R/is_lint_level.R

```diff
@@ -24,3 +24,23 @@ is_lint_level <- function(source_expression, level = c("expression", "file")) {
   required_key <- paste0(if (level == "file") "full_", "parsed_content")
   required_key %in% names(source_expression)
 }
+
+#' Determine whether a linter needs to run for a given source_expression level
+#'
+#' Used by [lint()] to avoid unneccessary calls to linters.
+#'
+#' @param linter A linter.
+#' @param level Which level of expression is being tested? `"expression"`
+#'   means an individual expression, while `"file"` means all expressions
+#'   in the current file are available.
+#'
+#' @keywords internal
+#' @noRd
+is_linter_level <- function(linter, level = c("expression", "file")) {
+  linter_level <- attr(linter, "linter_level", exact = TRUE)
+  if (is.null(linter_level) || is.na(linter_level)) {
```

Comment:
`if` will give a bad error if `length(linter_level) != 1L`, worth adjusting? `if (is.null(linter_level) || (length(linter_level) == 1L && is.na(linter_level)))` or `if (is.null(linter_level) && identical(linter_level, NA))`

Reply:
I think we can assume that the attribute isn't "hacked in". `Linter()` already takes care of validation.